microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/commons/JavaCursorContextKind.java

```diff
@@ -0,0 +1,95 @@
+/*******************************************************************************
+* Copyright (c) 2023 Red Hat Inc. and others.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v. 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+* which is available at https://www.apache.org/licenses/LICENSE-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lsp4mp.commons;
+
+/**
+ * Represents the context where the cursor is in a Java file.
+ */
+public enum JavaCursorContextKind {
+
+	/**
+	 * The cursor is in a file that does not have a root type declaration.
+	 */
+	IN_EMPTY_FILE(1),
+
+	/**
+	 * The cursor is before a type declaration body, either at the root of a file or
+	 * within another class. The cursor is before any annotations on the type.
+	 */
```

Comment:
Could you add please a sample for each enum like

```
|
class A
```

Having a sample is sometimes better than a description. If youhave the both, it is better again :)